File: remote/src/main/java/com/kelsos/mbrc/annotations/SettingsAction.java
package com.kelsos.mbrc.annotations;

import android.support.annotation.IntDef;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

public class SettingsAction {
  public static final int DELETE = 1;
  public static final int EDIT = 2;
  public static final int DEFAULT = 3;
  public static final int NEW = 4;

  private SettingsAction() {
    //no instance
  }

  @IntDef({
      DELETE,
      EDIT,
      DEFAULT,
      NEW

  })
  @Retention(RetentionPolicy.SOURCE)
  public @interface Action {

  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/annotations/SettingsAction.kt
package com.kelsos.mbrc.annotations

import android.support.annotation.IntDef

object SettingsAction {
    const val DELETE = 1
    const val EDIT = 2
    const val DEFAULT = 3
    const val NEW = 4

    @IntDef(DELETE.toLong(), EDIT.toLong(), DEFAULT.toLong(), NEW.toLong())
    @Retention(AnnotationRetention.SOURCE)
    annotation class Action
}
